adafruit_requests.py

```diff
@@ -399,10 +447,31 @@ def _send(socket: SocketType, data: bytes):
                 # Not EAGAIN; that was already handled.
                 raise OSError(errno.EIO)
             total_sent += sent
+        return total_sent
 
     def _send_as_bytes(self, socket: SocketType, data: str):
         return self._send(socket, bytes(data, "utf-8"))
 
+    def _send_boundary_objects(self, socket: SocketType, boundary_objects: Any):
+        for boundary_object in boundary_objects:
+            if isinstance(boundary_object, str):
+                self._send_as_bytes(socket, boundary_object)
+            else:
+                chunk_size = 32
```

Comment:
This section is assuming the file is opened in binary mode, should probably add something to allow both

Reply:
Hmm, I assumed that reading in bytes mode was a requirement because all of the examples I found and referenced used that. 

Looking further into it now I found that the requests docs here https://requests.readthedocs.io/en/latest/user/quickstart/#post-a-multipart-encoded-file contain a red warning that binary mode is strongly recommended. 
![image](https://github.com/FoamyGuy/Adafruit_CircuitPython_Requests/assets/2406189/68622d1e-3e7a-4363-ab9f-0053f0e8026d)

I checked a few of the top results on google for "requests upload file" and all of the ones I saw used `rb` mode even a few that were .txt and .csv which could have supported text mode. 


While it does seem that CPython can work with a simple example of a .txt file using mode "r", I think I'm okay just not supporting that in the circuitpython version. It'll keep the code a little leaner and less complex, plus will force the adoption of the "strong recommendation" that requests documentation makes.

Reply:
Sounds good to me. I was going to look into this one you reviewed... We should still see if we can detect and error if it's not binary

Reply:
I assume it would result in some kind of encoding error, but I'm certain and don't know where it might raise from off the top of my head. 

It should also be possible to detect the difference with `type()` (or `isinstance()`)

```
>>> f = open("something.txt", "rb")
>>> type(f)
<class 'FileIO'>
>>> f.close()
>>> f = open("something.txt", "r")
>>> type(f)
<class 'TextIOWrapper'>
>>> f.close()

```

Reply:
I read the first byte. Seemed the safest way, incase someone built a different file handler class